robot/EDA/resources/RelationshipsSettingsPageObject.py

```diff
@@ -42,3 +79,159 @@ def update_reciprocal_method_value(self,**kwargs):
+            elif str(field).lower() == "female":
+                locator = eda_lex_locators["eda_settings_relationships"]["new_settings"].format(4,str(field).lower(),value)
+                self.selenium.page_should_contain_element(locator)
+            elif str(field).lower() == "male":
+                locator = eda_lex_locators["eda_settings_relationships"]["new_settings"].format(3,str(field).lower(),value)
+                self.selenium.page_should_contain_element(locator)
+            else:
+                locator = eda_lex_locators["eda_settings_relationships"]["new_settings"].format(2,str(field).lower(),value)
+                self.selenium.page_should_contain_element(locator)
+
+    def verify_reciprocal_setting_edit_mode(self,**kwargs):
+        """ This method will validate the new reciprocal setting added recently which is always in
+            in last row in edit mdoe. Pass the name of the reciprocal relationship and its value as
+            keyword args and this method will assign constant numeric value against each setting
+            and validates the corresponding value
+        """
+        for field,value in kwargs.items():
+            if str(field).lower() == "name":
+                locator = eda_lex_locators["eda_settings_relationships"]["new_setting_edit"].format(5,str(field).lower(),field,value)
+                self.selenium.page_should_contain_element(locator)
+            elif str(field).lower() == "female":
+                locator = eda_lex_locators["eda_settings_relationships"]["new_setting_edit"].format(4,str(field).lower(),field,value)
+                self.selenium.page_should_contain_element(locator)
+            elif str(field).lower() == "male":
+                locator = eda_lex_locators["eda_settings_relationships"]["new_setting_edit"].format(3,str(field).lower(),field,value)
+                self.selenium.page_should_contain_element(locator)
+            else:
+                locator = eda_lex_locators["eda_settings_relationships"]["new_setting_edit"].format(2,str(field).lower(),field,value)
+                self.selenium.page_should_contain_element(locator)
+
+    def verify_settings_count(self, settings_count, actual_count, expected):
+        """ This method will compare the reciprocal relationship settings count. It accepts initial
+            count, updated count and expected status as arguments
+        """
+        if str(expected).lower() == "true":
+            no_of_records = actual_count - settings_count
+            self.builtin.log(no_of_records)
+            if no_of_records > 0:
+                return int(actual_count)
+            else:
+                raise Exception("The reciprocal settings are not updated")
+        else:
+            if actual_count == settings_count:
+                no_of_records = actual_count - settings_count
+                self.builtin.log(no_of_records)
+                return int(actual_count)
```

Comment:
If the counts are equal, why are you calculating the difference and returning it ?

Reply:
This is to maintain the uniformity in logging mainly to help readability. I am logging the difference in records but only returning the updated count in both cases which is needed for my test validation